pkg/component/controller/konnectivity.go

```diff
@@ -116,23 +116,24 @@ func (k *Konnectivity) defaultArgs() stringmap.StringMap {
 		logrus.Errorf("failed to fetch server ID for konnectivity-server")
 	}
 	return stringmap.StringMap{
-		"--uds-name":                filepath.Join(k.K0sVars.KonnectivitySocketDir, "konnectivity-server.sock"),
-		"--cluster-cert":            filepath.Join(k.K0sVars.CertRootDir, "server.crt"),
-		"--cluster-key":             filepath.Join(k.K0sVars.CertRootDir, "server.key"),
-		"--kubeconfig":              k.K0sVars.KonnectivityKubeConfigPath,
-		"--mode":                    "grpc",
-		"--server-port":             "0",
-		"--agent-port":              fmt.Sprintf("%d", k.clusterConfig.Spec.Konnectivity.AgentPort),
-		"--admin-port":              fmt.Sprintf("%d", k.clusterConfig.Spec.Konnectivity.AdminPort),
-		"--agent-namespace":         "kube-system",
-		"--agent-service-account":   "konnectivity-agent",
-		"--authentication-audience": "system:konnectivity-server",
-		"--logtostderr":             "true",
-		"--stderrthreshold":         "1",
-		"--v":                       k.LogLevel,
-		"--enable-profiling":        "false",
-		"--server-id":               serverID,
-		"--proxy-strategies":        "destHost,default",
+		"--uds-name":                 filepath.Join(k.K0sVars.KonnectivitySocketDir, "konnectivity-server.sock"),
+		"--cluster-cert":             filepath.Join(k.K0sVars.CertRootDir, "server.crt"),
+		"--cluster-key":              filepath.Join(k.K0sVars.CertRootDir, "server.key"),
+		"--kubeconfig":               k.K0sVars.KonnectivityKubeConfigPath,
+		"--mode":                     "grpc",
+		"--server-port":              "0",
+		"--agent-port":               fmt.Sprintf("%d", k.clusterConfig.Spec.Konnectivity.AgentPort),
+		"--admin-port":               fmt.Sprintf("%d", k.clusterConfig.Spec.Konnectivity.AdminPort),
+		"--agent-namespace":          "kube-system",
+		"--agent-service-account":    "konnectivity-agent",
+		"--authentication-audience":  "system:konnectivity-server",
+		"--logtostderr":              "true",
+		"--stderrthreshold":          "1",
+		"--v":                        k.LogLevel,
+		"--enable-profiling":         "false",
+		"--delete-existing-uds-file": "true",
```

Comment:
Should this `--delete-existing-uds-file` be here?

Reply:
It fixes the konnectivity restart issue on an unclean shutdown (#2218).